linkerd/app/outbound/src/lib.rs

```diff
@@ -156,6 +157,7 @@ impl Config {
                     }))
                     .push(observability.clone())
                     .push(identity_headers.clone())
+                    .push(http::overwrite_authority::layer(endpoint::FromMetadata,  vec![HOST.as_str(), CANONICAL_DST_HEADER]))
```

Comment:
Why does this need `FromMetadata`? hasn't the endpoint already been constructed at this point?

Reply:
Yeah, I added an `ExtractAuthority` impl to `FromMetadata`. Does that introduce tangling of concepts? 

```
impl ExtractAuthority<Target<HttpEndpoint>> for FromMetadata {
    fn extract(&self, target: &Target<HttpEndpoint>) -> Option<Authority> {
        target.inner.metadata.authority_override().cloned()
    }
}
```

Reply:
I sketched out an alternate approach here https://github.com/linkerd/linkerd2-proxy/pull/481/files (note that this is based on #480)